openmc/mgxs/library.py

```diff
@@ -717,3 +721,292 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+            Multi-Group Cross Section File that is ready to be printed to the
+            file of choice by the user.
+
+        Raises
+        ------
+        ValueError
+            When the Library object is initialized with insufficient types of
+            cross sections for the Library.
+
+        See also
+        --------
+        Library.dump_to_file(mgxs_lib, filename, directory)
+
+        """
+
+        # Check to ensure the Library contains the correct
+        # multi-group cross section types
+        self.check_library_for_openmc_mgxs()
+
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+        if domain_names is not None:
+            cv.check_iterable_type('domain_names', domain_names, basestring)
+        if xs_ids is not None:
+            if isinstance(xs_ids, basestring):
+                # If we only have a string lets convert it now to a list
+                # of strings.
+                xs_ids = [xs_ids for i in range(len(self.domains))]
+            else:
+                cv.check_iterable_type('xs_ids', xs_ids, basestring)
+        else:
+            xs_ids = ['1m' for i in range(len(self.domains))]
+
+        # If gathering material-specific data, set the xs_type to macro
+        if not self.by_nuclide:
+            xs_type = 'macro'
+
+        # Initialize file
+        mgxs_file = openmc.MGXSLibrary(self.energy_groups)
+
+        # Set the scattering order as isotropic until
+        # support for higher orders are included in openmc.mgxs
+        order = 0
+
+        # Build storage for our XSdata objects
+        xsdatas = []
+
+        for i, domain in enumerate(self.domains):
+            if self.by_nuclide:
+                nuclides = list(domain.get_all_nuclides().keys())
+            else:
+                nuclides = ['total']
+            for nuclide in nuclides:
+                # Build & add metadata to XSdata object
+                if domain_names is None:
+                    name = 'set' + str(i + 1)
+                else:
+                    name = domain_names[i]
+                if nuclide is not 'total':
+                    name += '_' + nuclide
+
+                xsdata = self.get_xsdata(domain, name, nuclide=nuclide,
+                                         xs_type=xs_type, xs_id=xs_ids[i],
+                                         order=order)
+
+                xsdatas.append(xsdata)
+
+        # Add XSdatas to file
+        mgxs_file.add_xsdatas(xsdatas)
+
+        return mgxs_file
+
+    def check_library_for_openmc_mgxs(self):
+        """This routine will check the MGXS Types within a Library
+        to ensure the MGXS types provided can be used to create
+        a MGXS Library for OpenMC's Multi-Group mode.
+
+        The rules to check include:
+        - Either total or transport should be present.
+          - Both can be available if one wants, but we should
+            use whatever corresponds to Library.correction (if P0: transport)
```

Comment:
For nested lists to render properly in the documentation, you need to surround the sub-lists [with blank lines](http://www.sphinx-doc.org/en/stable/rest.html#lists-and-quote-like-blocks).